Add example for "Deploy the Splunk Collector in agent mode and disable using the host network" (#1361)
signalfx/splunk-otel-collector-chart: examples/collector-agent-host-network-disabled/README.md

# Example of chart configuration

## Deploy the Splunk Collector in agent mode and disable using the host network

- This configuration will install the collector as an agent deployment only.

> - Disabling this value will affect monitoring of some control plane components.

Reviewer: What is `this value` in this context?

Author: Some metrics will be missing, but I don't think we know which ones will be missing. I'll see if I can use the tests in this PR to determine which metrics will be missing and add them here.

Reviewer 2: Paulo is asking what "this value" is literally as in - which toggle is being set to false?

Reviewer 2: to make this more obvious to me (hopefully I am representing well what Paulo is saying):
```suggestion
- Setting `agent.hostnetwork` to false will affect monitoring of some control plane components.
```

> - Enabling the agent service is recommended and also done in this example. Kubernetes services will be deployed to use transmit collector related data.

Reviewer: I'm not sure about the wording here, but, making a suggestion to start a conversation about it:
```suggestion
- Enabling the agent service, as done in this example, is recommended. In this case Kubernetes services will be deployed to transmit collector related data.
```

# Example of chart configuration

## Deploy the Splunk Collector in agent mode and disable using the host network

- This configuration will install the collector as an agent deployment only.
- Disabling this value will affect monitoring of some control plane components.

>   - TODO: Add metrics that would be missing

Reviewer 3: Host network metrics likely will be reported incorrectly. We need to investigate that and document it before merging
